app/src/main/java/in/co/rajkumaar/amritarepo/aumsV2/activities/HomeActivity.java

```diff
@@ -42,9 +44,35 @@ protected void onCreate(Bundle savedInstanceState) {
 
         getSupportActionBar().setDisplayHomeAsUpEnabled(true);
 
-        name.setText(preferences.getString("name", "N/A"));
-        user_name.setText(preferences.getString("username", "N/A"));
-        e_mail.setText(preferences.getString("email", "N/A"));
+        Encryption enc = null;
+        String rmName = null;
+        String rmEmail = null;
+        String rmUsername = null;
+        try {
+            rmName = preferences.getString("name", "N/A");
+            rmEmail = preferences.getString("email", "N/A");
+            rmUsername = preferences.getString("username", "N/A");
+
+            enc = new Encryption(HomeActivity.this, "aums-lite");
+            enc.generateSecretKey();
```

Comment:
Can you call this from the constructor itself? Instead of writing everywhere after creating the encryption object.